Codecademy/gamut
# chore: upgrade typescript

### Overview


Updates Typescript to 5.1.3


### PR Checklist

- [x] Related to JIRA ticket: [web-2378]

#### Testing Instructions



* Check out the PR locally
* Confirm all the checks pass and there are no errors
* Click around storybook + confirm everything looks fab
* Rinse + repeat for the monorepo + monolith PRS
* ...Profit!

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/36142)  | [Monolith Env](https://github.com/codecademy-engineering/Codecademy/pull/36142) |
| Monoreop       | [Portal PR](https://github.com/codecademy-engineering/mono/pull/2790)  |   Head to the monorepo PR! |
